### PR TITLE
Refactored generate_load: Make resources spawn in parallel

### PR DESCRIPTION
**Pull request by ntymtsiv**
Next resources are created in parallel: vms, vm snapshots, images
and volumes. It makes generate_load work more than in two times
faster. Also this resources have timeout, exceeded which will
raise Runtime error.

### CODE REVIEW

**Comment by realsystem** [comment]
+1

File: devlab/tests/generate_load.py [hunk 1]
```diff
@@ -280,37 +299,92 @@ def get_parameters_for_vm_creating(_vm):
...
-            snp = self.glanceclient.images.get(self.get_image_id(snapshot['image_name']))
-            while snp.status != 'active':
-                time.sleep(2)
-                snp = self.glanceclient.images.get(self.get_image_id(snapshot['image_name']))
-                if snp.status == 'error':
-                    return None
+            snp = self.glanceclient.images.get(self.get_image_id(
+                snapshot['image_name']))
+            snp_ids.append(snp.id)
+        for i in range(TIMEOUT):
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]: I see `for i in range(TIMEOUT)` loop for the 4th time. Please consider refactoring, keep [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself).

**Comment by roman-verchikov** [comment]
Please rebase and fix inline

File: devlab/tests/generate_load.py [hunk 2]
```diff
@@ -113,21 +131,10 @@ def get_user_tenant_roles(self, user):
...
-        while vlm.status != 'available' and vlm.status != 'in-use':
-            time.sleep(2)
-            vlm = self.cinderclient.volumes.get(
-                self.get_volume_id(volume_name))
-            if vlm.status == 'error':
-                    return None
+        srv = self.novaclient.servers.get(srv)
+        if srv.status == 'ACTIVE':
+            return True
+        return False
```
Comment (roman-verchikov) [inline devlab/tests/generate_load.py]:
```
if srv.status == 'ACTIVE':
      return True
return False
```
can be replaced with one line:
```
return srv.status == 'ACTIVE'
```
Less code is better code.

**Comment by roman-verchikov** [comment]
please rebase

**Comment by roman-verchikov** [comment]
+1, can be merged.

**Pull request merged**